### ISSUE TITLE
Build error during `cargo image --example-payload` (`could not compile quote`)

### ISSUE THREAD

**Issue by dimakuv** [confidential-containers/td-shim#549]
I get this error on the latest `main` commit:
```sh
~/td-shim$ cargo image --example-payload --release

warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /home/sdp/td-shim/library/ring/Cargo.toml
workspace: /home/sdp/td-shim/Cargo.toml
    Updating crates.io index
  Downloaded clap_lex v0.4.1
  Downloaded concolor-override v1.0.0
  Downloaded anstyle v0.3.5
  Downloaded concolor-query v0.3.3
  Downloaded clap v4.2.0
  Downloaded clap_builder v4.2.0
  Downloaded utf8parse v0.2.1
  Downloaded xshell v0.2.3
  Downloaded is-terminal v0.4.6
  Downloaded clap_derive v4.2.0
  Downloaded xshell-macros v0.2.3
  Downloaded anstyle-parse v0.1.1
  Downloaded anstream v0.2.6
  Downloaded rustix v0.37.4                                                                                                     Downloaded linux-raw-sys v0.3.0                                                                                               Downloaded 15 crates (1.6 MB) in 0.48s                                                                                         Compiling libc v0.2.140                                                                                                       Compiling io-lifetimes v1.0.9                                                                                                 Compiling proc-macro2 v1.0.54
   Compiling rustix v0.37.4
   Compiling bitflags v1.3.2
   Compiling quote v1.0.26
   Compiling linux-raw-sys v0.3.0
   Compiling unicode-ident v1.0.8
   Compiling utf8parse v0.2.1
   Compiling anstyle v0.3.5
   Compiling concolor-query v0.3.3
   Compiling concolor-override v1.0.0
   Compiling heck v0.4.1
   Compiling clap_lex v0.4.1
   Compiling strsim v0.10.0
   Compiling anyhow v1.0.70
   Compiling xshell-macros v0.2.3
   Compiling once_cell v1.17.1
   Compiling lazy_static v1.4.0
   Compiling anstyle-parse v0.1.1
error: could not compile `quote`

Caused by:
  process didn't exit successfully: `rustc --crate-name build_script_build --edition=2018 /home/sdp/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.26/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off --cfg 'feature="default"' --cfg 'feature="proc-macro"' -C metadata=ce7ea5d0f47ad962 -C extra-filename=-ce7ea5d0f47ad962 --out-dir /home/sdp/td-shim/target/release/build/quote-ce7ea5d0f47ad962 -L dependency=/home/sdp/td-shim/target/release/deps --cap-lints allow` (signal: 7, SIGBUS: access to undefined memory)
warning: build failed, waiting for other jobs to finish...
```

Some context:
```sh
$ rustc --version

rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
```

TD-Shim version:
```sh
~/td-shim$ git show

commit dbce7bbe36ab6a0538588db2ba29553b63c9d287 (HEAD -> main, origin/main, origin/HEAD)
Author: Jiaqi Gao <jiaqi.gao@intel.com>
Date:   Wed Mar 22 09:25:04 2023 -0400

    CI: use `cargo image` to build td-shim image

    Signed-off-by: Jiaqi Gao <jiaqi.gao@intel.com>
```

**Comment by dimakuv** [comment]
Note: building TD-Shim, the example payload and the linking them together *manually* works fine. I used `cargo xbuild ...` and `cargo run -p td-shim-tools ...` for this.

Thus, only the `cargo image` way seems to be broken.

**Comment by gaojiaqi7** [comment]
hi @dimakuv , I can't produce the error in my development environment and CI, could you please clean it up and try again?

**Comment by dimakuv** [comment]
Ok, in a clean environment it built successfully.

Apparently, the error before happened because I did this:
1. First checked out `v0.4.0` and built it.
2. But it didn't work on my TDX software stack, so I deleted the `target/` dir (thinking that's enough to clean up the environment).
3. Checked out `main` and built it. Error happened at this stage.

Most probably, [this intermediate step](https://github.com/confidential-containers/td-shim#before-build) screwed up the environment, and the `main` build process tried to use submodules from `v0.4.0`? Something like this, I guess.

Anyway, this problem is solved, closing it.
